[PDI-682] - Replace StringBuffer with StringBuilder for performance

mbatchelor opened the pull request:
[CHECKSTYLE] - Fix checkstyle on non-test classes

lucboudreau commented:
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
3. ant -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/compatibility/Row.java
  core/src/org/pentaho/di/compatibility/Value.java
  core/src/org/pentaho/di/core/BaseRowSet.java
  core/src/org/pentaho/di/core/CheckResult.java
  core/src/org/pentaho/di/core/Const.java
  core/src/org/pentaho/di/core/KettleClientEnvironment.java
  core/src/org/pentaho/di/core/Result.java
  core/src/org/pentaho/di/core/ResultFile.java
  core/src/org/pentaho/di/core/TimedRow.java
  core/src/org/pentaho/di/core/database/BaseDatabaseMeta.java
  core/src/org/pentaho/di/core/database/CacheDatabaseMeta.java
  core/src/org/pentaho/di/core/database/ConnectionPoolUtil.java
  core/src/org/pentaho/di/core/database/DB2DatabaseMeta.java
  core/src/org/pentaho/di/core/database/Database.java
  core/src/org/pentaho/di/core/database/DatabaseFactory.java
  core/src/org/pentaho/di/core/database/DatabaseMeta.java
  core/src/org/pentaho/di/core/database/Exasol4DatabaseMeta.java
  core/src/org/pentaho/di/core/database/HypersonicDatabaseMeta.java
  core/src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta.java
  core/src/org/pentaho/di/core/database/MonetDBDatabaseMeta.java
  core/src/org/pentaho/di/core/database/NeoviewDatabaseMeta.java
  core/src/org/pentaho/di/core/database/OracleDatabaseMeta.java
  core/src/org/pentaho/di/core/database/OracleRDBDatabaseMeta.java
  core/src/org/pentaho/di/core/database/SqlCommentScrubber.java
  core/src/org/pentaho/di/core/database/VectorWiseDatabaseMeta.java
  core/src/org/pentaho/di/core/database/util/DatabaseUtil.java
  core/src/org/pentaho/di/core/exception/KettleMissingPluginsException.java
  core/src/org/pentaho/di/core/extension/ExtensionPointMap.java
  core/src/org/pentaho/di/core/extension/KettleExtensionPoint.java
  core/src/org/pentaho/di/core/fileinput/FileInputList.java
  core/src/org/pentaho/di/core/jdbc/ThinResultSet.java
  core/src/org/pentaho/di/core/jdbc/ThinUtil.java
  core/src/org/pentaho/di/core/logging/BufferChangedListener.java
  core/src/org/pentaho/di/core/logging/KettleLogLayout.java
  core/src/org/pentaho/di/core/logging/LogMessage.java
  core/src/org/pentaho/di/core/logging/LoggingBuffer.java
  core/src/org/pentaho/di/core/logging/LoggingRegistry.java
  core/src/org/pentaho/di/core/plugins/BasePluginTypeExposer.java
  core/src/org/pentaho/di/core/plugins/KettleURLClassLoader.java
  core/src/org/pentaho/di/core/plugins/PluginRegistry.java
  core/src/org/pentaho/di/core/row/ValueDataUtil.java
  core/src/org/pentaho/di/core/row/ValueMetaAndData.java
  core/src/org/pentaho/di/core/row/value/ValueMetaBase.java
  core/src/org/pentaho/di/core/row/value/ValueMetaInternetAddress.java
  core/src/org/pentaho/di/core/row/value/ValueMetaTimestamp.java
  core/src/org/pentaho/di/core/util/SortedFileOutputStream.java
  core/src/org/pentaho/di/core/util/StringUtil.java
  core/src/org/pentaho/di/core/util/UUID4Util.java
  core/src/org/pentaho/di/core/vfs/KettleVFS.java
  core/src/org/pentaho/di/core/xml/XMLHandler.java
  core/src/org/pentaho/di/i18n/GlobalMessages.java
  core/src/org/pentaho/di/repository/RepositoryDirectory.java
  core/src/org/pentaho/di/www/WebResult.java
  core/test-src/org/pentaho/di/core/ConstTest.java
  engine/src/org/pentaho/di/cluster/ClusterSchema.java
  engine/src/org/pentaho/di/cluster/SlaveServer.java
  engine/src/org/pentaho/di/core/HTTPProtocol.java
  engine/src/org/pentaho/di/core/KettleEnvironment.java
  engine/src/org/pentaho/di/core/NotePadMeta.java
  engine/src/org/pentaho/di/core/attributes/metastore/EmbeddedMetaStore.java
  engine/src/org/pentaho/di/core/attributes/metastore/JsonElement.java
  engine/src/org/pentaho/di/core/attributes/metastore/JsonElementType.java
  engine/src/org/pentaho/di/core/auth/core/AuthenticationManager.java
  engine/src/org/pentaho/di/core/gui/SwingDirectGC.java
  engine/src/org/pentaho/di/core/lifecycle/KettleLifecycleSupport.java
  engine/src/org/pentaho/di/core/logging/BaseLogTable.java
  engine/src/org/pentaho/di/core/logging/ChannelLogTable.java
  engine/src/org/pentaho/di/core/logging/JobEntryLogTable.java
  engine/src/org/pentaho/di/core/logging/JobLogTable.java
  engine/src/org/pentaho/di/core/logging/MetricsLogTable.java
  engine/src/org/pentaho/di/core/logging/PerformanceLogTable.java
  engine/src/org/pentaho/di/core/logging/StepLogTable.java
  engine/src/org/pentaho/di/core/logging/TransLogTable.java
  engine/src/org/pentaho/di/core/reflection/StringSearchResult.java
  engine/src/org/pentaho/di/core/reflection/StringSearcher.java
  engine/src/org/pentaho/di/core/util/JavaScriptUtils.java
  engine/src/org/pentaho/di/core/util/SerializationHelper.java
  engine/src/org/pentaho/di/imp/Import.java
  engine/src/org/pentaho/di/job/JobConfiguration.java
  engine/src/org/pentaho/di/job/JobExecutionConfiguration.java
  engine/src/org/pentaho/di/job/JobHopMeta.java
  engine/src/org/pentaho/di/job/JobMeta.java
  engine/src/org/pentaho/di/job/entries/abort/JobEntryAbort.java
  engine/src/org/pentaho/di/job/entries/addresultfilenames/JobEntryAddResultFilenames.java
  engine/src/org/pentaho/di/job/entries/checkdbconnection/JobEntryCheckDbConnections.java
  engine/src/org/pentaho/di/job/entries/checkfilelocked/JobEntryCheckFilesLocked.java
  engine/src/org/pentaho/di/job/entries/columnsexist/JobEntryColumnsExist.java
  engine/src/org/pentaho/di/job/entries/connectedtorepository/JobEntryConnectedToRepository.java
  engine/src/org/pentaho/di/job/entries/copyfiles/JobEntryCopyFiles.java
  engine/src/org/pentaho/di/job/entries/copymoveresultfilenames/JobEntryCopyMoveResultFilenames.java
  engine/src/org/pentaho/di/job/entries/createfile/JobEntryCreateFile.java
  engine/src/org/pentaho/di/job/entries/createfolder/JobEntryCreateFolder.java
  engine/src/org/pentaho/di/job/entries/delay/JobEntryDelay.java
  engine/src/org/pentaho/di/job/entries/deletefile/JobEntryDeleteFile.java
  engine/src/org/pentaho/di/job/entries/deletefiles/JobEntryDeleteFiles.java
  engine/src/org/pentaho/di/job/entries/deletefolders/JobEntryDeleteFolders.java
  engine/src/org/pentaho/di/job/entries/deleteresultfilenames/JobEntryDeleteResultFilenames.java
  engine/src/org/pentaho/di/job/entries/dostounix/JobEntryDosToUnix.java
  engine/src/org/pentaho/di/job/entries/eval/JobEntryEval.java
  engine/src/org/pentaho/di/job/entries/evalfilesmetrics/JobEntryEvalFilesMetrics.java
  engine/src/org/pentaho/di/job/entries/evaluatetablecontent/JobEntryEvalTableContent.java
  engine/src/org/pentaho/di/job/entries/exportrepository/JobEntryExportRepository.java
  engine/src/org/pentaho/di/job/entries/filecompare/JobEntryFileCompare.java
  engine/src/org/pentaho/di/job/entries/fileexists/JobEntryFileExists.java
  engine/src/org/pentaho/di/job/entries/filesexist/JobEntryFilesExist.java
  engine/src/org/pentaho/di/job/entries/folderisempty/JobEntryFolderIsEmpty.java
  engine/src/org/pentaho/di/job/entries/folderscompare/JobEntryFoldersCompare.java
  engine/src/org/pentaho/di/job/entries/ftp/JobEntryFTP.java
  engine/src/org/pentaho/di/job/entries/ftp/MVSFileParser.java
  engine/src/org/pentaho/di/job/entries/ftpdelete/JobEntryFTPDelete.java
  engine/src/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUT.java
  engine/src/org/pentaho/di/job/entries/ftpsget/FTPSConnection.java
  engine/src/org/pentaho/di/job/entries/ftpsget/JobEntryFTPSGet.java
  engine/src/org/pentaho/di/job/entries/ftpsput/JobEntryFTPSPUT.java
  engine/src/org/pentaho/di/job/entries/getpop/JobEntryGetPOP.java
  engine/src/org/pentaho/di/job/entries/getpop/MailConnection.java
  engine/src/org/pentaho/di/job/entries/http/JobEntryHTTP.java
  engine/src/org/pentaho/di/job/entries/job/JobEntryJob.java
  engine/src/org/pentaho/di/job/entries/mail/JobEntryMail.java
  engine/src/org/pentaho/di/job/entries/mailvalidator/JobEntryMailValidator.java
  engine/src/org/pentaho/di/job/entries/movefiles/JobEntryMoveFiles.java
  engine/src/org/pentaho/di/job/entries/msgboxinfo/JobEntryMsgBoxInfo.java
  engine/src/org/pentaho/di/job/entries/mssqlbulkload/JobEntryMssqlBulkLoad.java
  engine/src/org/pentaho/di/job/entries/mysqlbulkfile/JobEntryMysqlBulkFile.java
  engine/src/org/pentaho/di/job/entries/mysqlbulkload/JobEntryMysqlBulkLoad.java
  engine/src/org/pentaho/di/job/entries/pgpdecryptfiles/JobEntryPGPDecryptFiles.java
  engine/src/org/pentaho/di/job/entries/pgpencryptfiles/GPG.java
  engine/src/org/pentaho/di/job/entries/pgpencryptfiles/JobEntryPGPEncryptFiles.java
  engine/src/org/pentaho/di/job/entries/pgpverify/JobEntryPGPVerify.java
  engine/src/org/pentaho/di/job/entries/ping/JobEntryPing.java
  engine/src/org/pentaho/di/job/entries/sendnagiospassivecheck/JobEntrySendNagiosPassiveCheck.java
  engine/src/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java
  engine/src/org/pentaho/di/job/entries/sftp/JobEntrySFTP.java
  engine/src/org/pentaho/di/job/entries/sftpput/JobEntrySFTPPUT.java
  engine/src/org/pentaho/di/job/entries/shell/JobEntryShell.java
  engine/src/org/pentaho/di/job/entries/simpleeval/JobEntrySimpleEval.java
  engine/src/org/pentaho/di/job/entries/snmptrap/JobEntrySNMPTrap.java
  engine/src/org/pentaho/di/job/entries/special/JobEntrySpecial.java
  engine/src/org/pentaho/di/job/entries/sql/JobEntrySQL.java
  engine/src/org/pentaho/di/job/entries/success/JobEntrySuccess.java
  engine/src/org/pentaho/di/job/entries/syslog/JobEntrySyslog.java
  engine/src/org/pentaho/di/job/entries/tableexists/JobEntryTableExists.java
  engine/src/org/pentaho/di/job/entries/talendjobexec/JobEntryTalendJobExec.java
  engine/src/org/pentaho/di/job/entries/telnet/JobEntryTelnet.java
  engine/src/org/pentaho/di/job/entries/trans/JobEntryTrans.java
  engine/src/org/pentaho/di/job/entries/truncatetables/JobEntryTruncateTables.java
  engine/src/org/pentaho/di/job/entries/unzip/JobEntryUnZip.java
  engine/src/org/pentaho/di/job/entries/waitforfile/JobEntryWaitForFile.java
  engine/src/org/pentaho/di/job/entries/waitforsql/JobEntryWaitForSQL.java
  engine/src/org/pentaho/di/job/entries/webserviceavailable/JobEntryWebServiceAvailable.java
  engine/src/org/pentaho/di/job/entries/writetofile/JobEntryWriteToFile.java
  engine/src/org/pentaho/di/job/entries/writetolog/JobEntryWriteToLog.java
  engine/src/org/pentaho/di/job/entries/zipfile/JobEntryZipFile.java
  engine/src/org/pentaho/di/job/entry/JobEntryBase.java
  engine/src/org/pentaho/di/job/entry/JobEntryCopy.java
  engine/src/org/pentaho/di/job/entry/validator/EmailValidator.java
  engine/src/org/pentaho/di/job/entry/validator/IntegerValidator.java
  engine/src/org/pentaho/di/job/entry/validator/LongValidator.java
  engine/src/org/pentaho/di/kitchen/Kitchen.java
  engine/src/org/pentaho/di/pan/CommandLineOption.java
  engine/src/org/pentaho/di/pan/Pan.java
  engine/src/org/pentaho/di/partition/PartitionSchema.java
  engine/src/org/pentaho/di/repository/BaseRepositoryMeta.java
  engine/src/org/pentaho/di/repository/RepositoriesMeta.java
  engine/src/org/pentaho/di/repository/RepositoryCommonValidations.java
  engine/src/org/pentaho/di/repository/filerep/KettleFileRepository.java
  engine/src/org/pentaho/di/repository/filerep/KettleFileRepositoryMeta.java
  engine/src/org/pentaho/di/repository/kdr/KettleDatabaseRepository.java
  engine/src/org/pentaho/di/repository/kdr/KettleDatabaseRepositoryBase.java
  engine/src/org/pentaho/di/repository/kdr/KettleDatabaseRepositoryMeta.java
  engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryClusterSchemaDelegate.java
  engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryJobDelegate.java
  engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryJobEntryDelegate.java
  engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryPartitionSchemaDelegate.java
  engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryStepDelegate.java
  engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryTransDelegate.java
  engine/src/org/pentaho/di/resource/ResourceEntry.java
  engine/src/org/pentaho/di/resource/ResourceReference.java
  engine/src/org/pentaho/di/resource/SimpleResourceNaming.java
  engine/src/org/pentaho/di/trans/Trans.java
  engine/src/org/pentaho/di/trans/TransMeta.java
  engine/src/org/pentaho/di/trans/step/BaseStep.java
  engine/src/org/pentaho/di/trans/step/RemoteStep.java
  engine/src/org/pentaho/di/trans/step/StepErrorMeta.java
  engine/src/org/pentaho/di/trans/step/StepMeta.java
  engine/src/org/pentaho/di/trans/step/StepPartitioningMeta.java
  engine/src/org/pentaho/di/trans/steps/accessinput/AccessInput.java
  engine/src/org/pentaho/di/trans/steps/accessinput/AccessInputMeta.java
  engine/src/org/pentaho/di/trans/steps/accessoutput/AccessOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/addsequence/AddSequenceMeta.java
  engine/src/org/pentaho/di/trans/steps/analyticquery/AnalyticQueryMeta.java
  engine/src/org/pentaho/di/trans/steps/autodoc/AutoDocMeta.java
  engine/src/org/pentaho/di/trans/steps/blockuntilstepsfinish/BlockUntilStepsFinishMeta.java
  engine/src/org/pentaho/di/trans/steps/changefileencoding/ChangeFileEncodingMeta.java
  engine/src/org/pentaho/di/trans/steps/checksum/CheckSum.java
  engine/src/org/pentaho/di/trans/steps/checksum/CheckSumMeta.java
  engine/src/org/pentaho/di/trans/steps/clonerow/CloneRowMeta.java
  engine/src/org/pentaho/di/trans/steps/closure/ClosureGeneratorMeta.java
  engine/src/org/pentaho/di/trans/steps/columnexists/ColumnExistsMeta.java
  engine/src/org/pentaho/di/trans/steps/combinationlookup/CombinationLookup.java
  engine/src/org/pentaho/di/trans/steps/combinationlookup/CombinationLookupMeta.java
  engine/src/org/pentaho/di/trans/steps/constant/ConstantMeta.java
  engine/src/org/pentaho/di/trans/steps/creditcardvalidator/CreditCardValidatorMeta.java
  engine/src/org/pentaho/di/trans/steps/csvinput/CsvInputMeta.java
  engine/src/org/pentaho/di/trans/steps/cubeinput/CubeInput.java
  engine/src/org/pentaho/di/trans/steps/cubeinput/CubeInputMeta.java
  engine/src/org/pentaho/di/trans/steps/cubeoutput/CubeOutput.java
  engine/src/org/pentaho/di/trans/steps/cubeoutput/CubeOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/databasejoin/DatabaseJoinMeta.java
  engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookupMeta.java
  engine/src/org/pentaho/di/trans/steps/databaselookup/readallcache/Index.java
  engine/src/org/pentaho/di/trans/steps/datagrid/DataGridMeta.java
  engine/src/org/pentaho/di/trans/steps/dbproc/DBProc.java
  engine/src/org/pentaho/di/trans/steps/dbproc/DBProcMeta.java
  engine/src/org/pentaho/di/trans/steps/delay/DelayMeta.java
  engine/src/org/pentaho/di/trans/steps/delete/Delete.java
  engine/src/org/pentaho/di/trans/steps/delete/DeleteMeta.java
  engine/src/org/pentaho/di/trans/steps/denormaliser/DenormaliserMeta.java
  engine/src/org/pentaho/di/trans/steps/dimensionlookup/DimensionLookup.java
  engine/src/org/pentaho/di/trans/steps/dimensionlookup/DimensionLookupMeta.java
  engine/src/org/pentaho/di/trans/steps/dynamicsqlrow/DynamicSQLRowMeta.java
  engine/src/org/pentaho/di/trans/steps/edi2xml/Edi2XmlMeta.java
  engine/src/org/pentaho/di/trans/steps/excelinput/ExcelInputMeta.java
  engine/src/org/pentaho/di/trans/steps/exceloutput/ExcelOutput.java
  engine/src/org/pentaho/di/trans/steps/exceloutput/ExcelOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java
  engine/src/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStepMeta.java
  engine/src/org/pentaho/di/trans/steps/execprocess/ExecProcess.java
  engine/src/org/pentaho/di/trans/steps/execprocess/ExecProcessMeta.java
  engine/src/org/pentaho/di/trans/steps/execsqlrow/ExecSQLRowMeta.java
  engine/src/org/pentaho/di/trans/steps/fieldschangesequence/FieldsChangeSequenceMeta.java
  engine/src/org/pentaho/di/trans/steps/fileexists/FileExistsMeta.java
  engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStepMeta.java
  engine/src/org/pentaho/di/trans/steps/fileinput/IBaseFileInputReader.java
  engine/src/org/pentaho/di/trans/steps/fileinput/IBaseFileInputStepControl.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMeta.java
  engine/src/org/pentaho/di/trans/steps/filelocked/FileLockedMeta.java
  engine/src/org/pentaho/di/trans/steps/filestoresult/FilesToResultMeta.java
  engine/src/org/pentaho/di/trans/steps/filterrows/FilterRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/fixedinput/FixedFileInputField.java
  engine/src/org/pentaho/di/trans/steps/fixedinput/FixedInputMeta.java
  engine/src/org/pentaho/di/trans/steps/flattener/FlattenerMeta.java
  engine/src/org/pentaho/di/trans/steps/formula/FormulaMeta.java
  engine/src/org/pentaho/di/trans/steps/fuzzymatch/FuzzyMatchMeta.java
  engine/src/org/pentaho/di/trans/steps/getfilenames/GetFileNames.java
  engine/src/org/pentaho/di/trans/steps/getfilenames/GetFileNamesMeta.java
  engine/src/org/pentaho/di/trans/steps/getfilesrowscount/GetFilesRowsCount.java
  engine/src/org/pentaho/di/trans/steps/getfilesrowscount/GetFilesRowsCountData.java
  engine/src/org/pentaho/di/trans/steps/getfilesrowscount/GetFilesRowsCountMeta.java
  engine/src/org/pentaho/di/trans/steps/getrepositorynames/GetRepositoryNamesMeta.java
  engine/src/org/pentaho/di/trans/steps/getslavesequence/GetSlaveSequenceMeta.java
  engine/src/org/pentaho/di/trans/steps/getsubfolders/GetSubFolders.java
  engine/src/org/pentaho/di/trans/steps/getsubfolders/GetSubFoldersMeta.java
  engine/src/org/pentaho/di/trans/steps/gettablenames/GetTableNamesMeta.java
  engine/src/org/pentaho/di/trans/steps/getvariable/GetVariableMeta.java
  engine/src/org/pentaho/di/trans/steps/groupby/GroupBy.java
  engine/src/org/pentaho/di/trans/steps/groupby/GroupByMeta.java
  engine/src/org/pentaho/di/trans/steps/http/HTTP.java
  engine/src/org/pentaho/di/trans/steps/http/HTTPMeta.java
  engine/src/org/pentaho/di/trans/steps/httppost/HTTPPOST.java
  engine/src/org/pentaho/di/trans/steps/httppost/HTTPPOSTMeta.java
  engine/src/org/pentaho/di/trans/steps/ifnull/IfNullMeta.java
  engine/src/org/pentaho/di/trans/steps/janino/JaninoMeta.java
  engine/src/org/pentaho/di/trans/steps/janino/JaninoMetaInjection.java
  engine/src/org/pentaho/di/trans/steps/javafilter/JavaFilterMeta.java
  engine/src/org/pentaho/di/trans/steps/jobexecutor/JobExecutorMeta.java
  engine/src/org/pentaho/di/trans/steps/jobexecutor/JobExecutorParameters.java
  engine/src/org/pentaho/di/trans/steps/joinrows/JoinRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/ldapinput/LDAPConnection.java
  engine/src/org/pentaho/di/trans/steps/ldapinput/LDAPInputMeta.java
  engine/src/org/pentaho/di/trans/steps/ldapoutput/LDAPOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/ldifinput/LDIFInput.java
  engine/src/org/pentaho/di/trans/steps/ldifinput/LDIFInputMeta.java
  engine/src/org/pentaho/di/trans/steps/loadfileinput/LoadFileInput.java
  engine/src/org/pentaho/di/trans/steps/loadfileinput/LoadFileInputMeta.java
  engine/src/org/pentaho/di/trans/steps/mail/Mail.java
  engine/src/org/pentaho/di/trans/steps/mail/MailMeta.java
  engine/src/org/pentaho/di/trans/steps/mailinput/MailInput.java
  engine/src/org/pentaho/di/trans/steps/mailinput/MailInputMeta.java
  engine/src/org/pentaho/di/trans/steps/mailvalidator/MailValidatorMeta.java
  engine/src/org/pentaho/di/trans/steps/mapping/MappingIODefinition.java
  engine/src/org/pentaho/di/trans/steps/mapping/MappingMeta.java
  engine/src/org/pentaho/di/trans/steps/mapping/MappingParameters.java
  engine/src/org/pentaho/di/trans/steps/mappinginput/MappingInputMeta.java
  engine/src/org/pentaho/di/trans/steps/memgroupby/MemoryGroupByMeta.java
  engine/src/org/pentaho/di/trans/steps/mergejoin/MergeJoinMeta.java
  engine/src/org/pentaho/di/trans/steps/mergerows/MergeRows.java
  engine/src/org/pentaho/di/trans/steps/mergerows/MergeRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/metainject/MetaInjectMeta.java
  engine/src/org/pentaho/di/trans/steps/mondrianinput/MondrianHelper.java
  engine/src/org/pentaho/di/trans/steps/mondrianinput/MondrianInputMeta.java
  engine/src/org/pentaho/di/trans/steps/monetdbbulkloader/MonetDBBulkLoader.java
  engine/src/org/pentaho/di/trans/steps/monetdbbulkloader/MonetDBBulkLoaderMeta.java
  engine/src/org/pentaho/di/trans/steps/monetdbbulkloader/MonetDbVersion.java
  engine/src/org/pentaho/di/trans/steps/multimerge/MultiMergeJoinMeta.java
  engine/src/org/pentaho/di/trans/steps/mysqlbulkloader/MySQLBulkLoaderMeta.java
  engine/src/org/pentaho/di/trans/steps/normaliser/NormaliserMeta.java
  engine/src/org/pentaho/di/trans/steps/nullif/NullIfMeta.java
  engine/src/org/pentaho/di/trans/steps/numberrange/NumberRangeMeta.java
  engine/src/org/pentaho/di/trans/steps/numberrange/NumberRangeSet.java
  engine/src/org/pentaho/di/trans/steps/olapinput/OlapInputMeta.java
  engine/src/org/pentaho/di/trans/steps/orabulkloader/OraBulkDataOutput.java
  engine/src/org/pentaho/di/trans/steps/orabulkloader/OraBulkLoaderMeta.java
  engine/src/org/pentaho/di/trans/steps/parallelgzipcsv/ParGzipCsvInputMeta.java
  engine/src/org/pentaho/di/trans/steps/pentahoreporting/PentahoReportingOutput.java
  engine/src/org/pentaho/di/trans/steps/pentahoreporting/PentahoReportingOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/pgbulkloader/PGBulkLoader.java
  engine/src/org/pentaho/di/trans/steps/pgbulkloader/PGBulkLoaderMeta.java
  engine/src/org/pentaho/di/trans/steps/pgpdecryptstream/PGPDecryptStreamMeta.java
  engine/src/org/pentaho/di/trans/steps/pgpencryptstream/PGPEncryptStreamMeta.java
  engine/src/org/pentaho/di/trans/steps/prioritizestreams/PrioritizeStreamsMeta.java
  engine/src/org/pentaho/di/trans/steps/processfiles/ProcessFilesMeta.java
  engine/src/org/pentaho/di/trans/steps/propertyinput/PropertyInput.java
  engine/src/org/pentaho/di/trans/steps/propertyinput/PropertyInputMeta.java
  engine/src/org/pentaho/di/trans/steps/propertyoutput/PropertyOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/randomccnumber/RandomCCNumberGeneratorMeta.java
  engine/src/org/pentaho/di/trans/steps/randomvalue/RandomValue.java
  engine/src/org/pentaho/di/trans/steps/randomvalue/RandomValueMeta.java
  engine/src/org/pentaho/di/trans/steps/replacestring/ReplaceStringMeta.java
  engine/src/org/pentaho/di/trans/steps/rest/RestMeta.java
  engine/src/org/pentaho/di/trans/steps/rowgenerator/RowGeneratorMeta.java
  engine/src/org/pentaho/di/trans/steps/rowsfromresult/RowsFromResultMeta.java
  engine/src/org/pentaho/di/trans/steps/rssinput/RssInput.java
  engine/src/org/pentaho/di/trans/steps/rssinput/RssInputMeta.java
  engine/src/org/pentaho/di/trans/steps/rssoutput/RssOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/samplerows/SampleRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/sapinput/SapInputMeta.java
  engine/src/org/pentaho/di/trans/steps/sapinput/mock/SAPConnectionFactoryMock.java
  engine/src/org/pentaho/di/trans/steps/sapinput/sap/SAPConnectionFactory.java
  engine/src/org/pentaho/di/trans/steps/sasinput/SasInputField.java
  engine/src/org/pentaho/di/trans/steps/sasinput/SasInputMeta.java
  engine/src/org/pentaho/di/trans/steps/script/Script.java
  engine/src/org/pentaho/di/trans/steps/script/ScriptMeta.java
  engine/src/org/pentaho/di/trans/steps/scriptvalues_mod/ScriptValuesAddedFunctions.java
  engine/src/org/pentaho/di/trans/steps/scriptvalues_mod/ScriptValuesMetaMod.java
  engine/src/org/pentaho/di/trans/steps/scriptvalues_mod/ScriptValuesMod.java
  engine/src/org/pentaho/di/trans/steps/selectvalues/SelectMetadataChange.java
  engine/src/org/pentaho/di/trans/steps/selectvalues/SelectValuesMeta.java
  engine/src/org/pentaho/di/trans/steps/setvalueconstant/SetValueConstantMeta.java
  engine/src/org/pentaho/di/trans/steps/setvaluefield/SetValueFieldMeta.java
  engine/src/org/pentaho/di/trans/steps/setvariable/SetVariableMeta.java
  engine/src/org/pentaho/di/trans/steps/sftpput/SFTPPutMeta.java
  engine/src/org/pentaho/di/trans/steps/simplemapping/SimpleMappingMeta.java
  engine/src/org/pentaho/di/trans/steps/singlethreader/SingleThreaderMeta.java
  engine/src/org/pentaho/di/trans/steps/socketreader/SocketReaderMeta.java
  engine/src/org/pentaho/di/trans/steps/socketwriter/SocketWriter.java
  engine/src/org/pentaho/di/trans/steps/socketwriter/SocketWriterMeta.java
  engine/src/org/pentaho/di/trans/steps/sort/SortRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/sortedmerge/SortedMergeMeta.java
  engine/src/org/pentaho/di/trans/steps/splitfieldtorows/SplitFieldToRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/sql/ExecSQL.java
  engine/src/org/pentaho/di/trans/steps/sql/ExecSQLMeta.java
  engine/src/org/pentaho/di/trans/steps/sqlfileoutput/SQLFileOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/ssh/SSHMeta.java
  engine/src/org/pentaho/di/trans/steps/stepmeta/StepMetastructureMeta.java
  engine/src/org/pentaho/di/trans/steps/stepsmetrics/StepsMetricsMeta.java
  engine/src/org/pentaho/di/trans/steps/streamlookup/StreamLookup.java
  engine/src/org/pentaho/di/trans/steps/stringcut/StringCutMeta.java
  engine/src/org/pentaho/di/trans/steps/stringoperations/StringOperationsMeta.java
  engine/src/org/pentaho/di/trans/steps/switchcase/SwitchCaseMeta.java
  engine/src/org/pentaho/di/trans/steps/symmetriccrypto/secretkeygenerator/SecretKeyGeneratorMeta.java
  engine/src/org/pentaho/di/trans/steps/symmetriccrypto/symmetriccryptotrans/SymmetricCryptoTransMeta.java
  engine/src/org/pentaho/di/trans/steps/synchronizeaftermerge/SynchronizeAfterMerge.java
  engine/src/org/pentaho/di/trans/steps/synchronizeaftermerge/SynchronizeAfterMergeMeta.java
  engine/src/org/pentaho/di/trans/steps/syslog/SyslogMessageMeta.java
  engine/src/org/pentaho/di/trans/steps/systemdata/SystemDataMeta.java
  engine/src/org/pentaho/di/trans/steps/tableagilemart/TableAgileMart.java
  engine/src/org/pentaho/di/trans/steps/tablecompare/TableCompare.java
  engine/src/org/pentaho/di/trans/steps/tablecompare/TableCompareMeta.java
  engine/src/org/pentaho/di/trans/steps/tableexists/TableExistsMeta.java
  engine/src/org/pentaho/di/trans/steps/tableinput/TableInput.java
  engine/src/org/pentaho/di/trans/steps/tableinput/TableInputMeta.java
  engine/src/org/pentaho/di/trans/steps/tableoutput/TableOutput.java
  engine/src/org/pentaho/di/trans/steps/tableoutput/TableOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/terafast/TeraFast.java
  engine/src/org/pentaho/di/trans/steps/textfileinput/TextFileInputMeta.java
  engine/src/org/pentaho/di/trans/steps/textfileinput/TextFileInputMetaInjection.java
  engine/src/org/pentaho/di/trans/steps/textfileoutput/TextFileOutput.java
  engine/src/org/pentaho/di/trans/steps/textfileoutput/TextFileOutputMeta.java
  engine/src/org/pentaho/di/trans/steps/transexecutor/TransExecutorParameters.java
  engine/src/org/pentaho/di/trans/steps/uniquerows/UniqueRowsMeta.java
  engine/src/org/pentaho/di/trans/steps/uniquerowsbyhashset/UniqueRowsByHashSetMeta.java
  engine/src/org/pentaho/di/trans/steps/univariatestats/UnivariateStatsMeta.java
  engine/src/org/pentaho/di/trans/steps/update/UpdateMeta.java
  engine/src/org/pentaho/di/trans/steps/userdefinedjavaclass/UserDefinedJavaClassMeta.java
  engine/src/org/pentaho/di/trans/steps/validator/Validation.java
  engine/src/org/pentaho/di/trans/steps/validator/ValidatorMeta.java
  engine/src/org/pentaho/di/trans/steps/valuemapper/ValueMapperMeta.java
  engine/src/org/pentaho/di/trans/steps/webserviceavailable/WebServiceAvailableMeta.java
  engine/src/org/pentaho/di/trans/steps/webservices/WebService.java
  engine/src/org/pentaho/di/trans/steps/webservices/WebServiceMeta.java
  engine/src/org/pentaho/di/trans/steps/writetolog/WriteToLog.java
  engine/src/org/pentaho/di/trans/steps/writetolog/WriteToLogMeta.java
  engine/src/org/pentaho/di/trans/steps/xbaseinput/XBaseInput.java
  engine/src/org/pentaho/di/trans/steps/xbaseinput/XBaseInputMeta.java
  engine/src/org/pentaho/di/trans/steps/yamlinput/YamlInputField.java
  engine/src/org/pentaho/di/trans/steps/yamlinput/YamlInputMeta.java
  engine/src/org/pentaho/di/trans/steps/zipfile/ZipFileMeta.java
  engine/src/org/pentaho/di/www/AddJobServlet.java
  engine/src/org/pentaho/di/www/CarteSingleton.java
  engine/src/org/pentaho/di/www/GetStatusServlet.java
  engine/src/org/pentaho/di/www/SlaveServerConfig.java
  engine/src/org/pentaho/di/www/SlaveServerDetection.java
  engine/src/org/pentaho/di/www/SlaveServerJobStatus.java
  engine/src/org/pentaho/di/www/SlaveServerStatus.java
  engine/src/org/pentaho/di/www/SlaveServerTransStatus.java
  engine/src/org/pentaho/di/www/StopCarteServlet.java
  engine/src/org/pentaho/libformula/editor/function/FunctionDescription.java
  engine/test-src/org/pentaho/di/trans/steps/ssh/SSHMetaTest.java
  engine/test-src/org/pentaho/di/www/SlaveServerConfigTest.java
  ui/src/org/pentaho/di/core/dnd/DragAndDropContainer.java
  ui/src/org/pentaho/di/ui/core/database/dialog/SQLEditor.java
  ui/src/org/pentaho/di/ui/core/dialog/CheckResultDialog.java
  ui/src/org/pentaho/di/ui/core/dialog/EnterPrintDialog.java
  ui/src/org/pentaho/di/ui/core/dialog/EnterValueDialog.java
  ui/src/org/pentaho/di/ui/core/dialog/ErrorDialog.java
  ui/src/org/pentaho/di/ui/core/dialog/SQLStatementsDialog.java
  ui/src/org/pentaho/di/ui/core/widget/TableView.java
  ui/src/org/pentaho/di/ui/core/widget/TreeMemory.java
  ui/src/org/pentaho/di/ui/i18n/editor/Translator.java
  ui/src/org/pentaho/di/ui/i18n/editor/Translator2.java
  ui/src/org/pentaho/di/ui/job/entries/missing/MissingEntryDialog.java
  ui/src/org/pentaho/di/ui/repository/dialog/RepositoryImportProgressDialog.java
  ui/src/org/pentaho/di/ui/repository/kdr/KettleDatabaseRepositoryDialog.java
  ui/src/org/pentaho/di/ui/spoon/Spoon.java
  ui/src/org/pentaho/di/ui/spoon/dialog/DeleteMessageBox.java
  ui/src/org/pentaho/di/ui/spoon/job/JobGraph.java
  ui/src/org/pentaho/di/ui/spoon/trans/LogBrowser.java
  ui/src/org/pentaho/di/ui/spoon/trans/TransGraph.java
  ui/src/org/pentaho/di/ui/spoon/trans/TransPreviewDelegate.java
  ui/src/org/pentaho/di/ui/trans/steps/ivwloader/IngresVectorwiseLoaderDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/ldapoutput/LDAPOutputDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/missing/MissingTransDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/mondrianinput/MDXValuesHighlight.java
  ui/src/org/pentaho/di/ui/trans/steps/olapinput/MDXValuesHighlight.java
  ui/src/org/pentaho/di/ui/trans/steps/script/ScriptDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/script/ScriptHelp.java
  ui/src/org/pentaho/di/ui/trans/steps/script/ScriptHighlight.java
  ui/src/org/pentaho/di/ui/trans/steps/scriptvalues_mod/ScriptValuesHelp.java
  ui/src/org/pentaho/di/ui/trans/steps/scriptvalues_mod/ScriptValuesHighlight.java
  ui/src/org/pentaho/di/ui/trans/steps/scriptvalues_mod/ScriptValuesModDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/selectvalues/SelectValuesDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/tableinput/SQLValuesHighlight.java
  ui/src/org/pentaho/di/ui/trans/steps/terafast/TeraFastDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/userdefinedjavaclass/UserDefinedJavaClassDialog.java
  ui/src/org/pentaho/di/ui/trans/steps/userdefinedjavaclass/UserDefinedJavaClassHighlight.java
  ui/src/org/pentaho/libformula/ui/editor/LibFormulaEditor.java
  ui/src/org/pentaho/libformula/ui/editor/LibFormulaValuesHighlight.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File core/test-src/org/pentaho/di/core/ConstTest.java
1. Message: 'method def rcurly' have incorrect indentation level 0, expected level should be 2. at Line: 1583 Column: None

File engine/test-src/org/pentaho/di/trans/steps/ssh/SSHMetaTest.java
1. Message: ',' is preceded with whitespace. at Line: 126 Column: 76
2. Message: Using a static member import should be avoided - org.junit.Assert.assertEquals. at Line: 30 Column: None

File engine/test-src/org/pentaho/di/www/SlaveServerConfigTest.java
1. Message: Using a static member import should be avoided - org.junit.Assert.assertEquals. at Line: 24 Column: None
2. Message: Using a static member import should be avoided - org.junit.Assert.assertFalse. at Line: 25 Column: None
3. Message: Using a static member import should be avoided - org.junit.Assert.assertNotNull. at Line: 26 Column: None
4. Message: Using a static member import should be avoided - org.junit.Assert.assertNull. at Line: 27 Column: None
5. Message: Using a static member import should be avoided - org.junit.Assert.assertTrue. at Line: 28 Column: None









Unit test coverage change:
============
org.pentaho.di.core.database.DatabaseMeta:
 * instructionPercentCoverageChange :**-0.28%**
 * branchPercentCoverageChange :**-0.54%**
 * linePercentCoverageChange :**-0.22%**
 * complexityPercentCoverageChange :**-0.22%**
 * methodPercentCoverageChange :**-0.09%**
    
org.pentaho.di.compatibility.Value:
 * instructionPercentCoverageChange : +0.13%
 * branchPercentCoverageChange : +0.13%
 * linePercentCoverageChange : +0.15%
 * complexityPercentCoverageChange :**-0.05%**
 * methodPercentCoverageChange : +0.08%
    
org.pentaho.di.core.Const:
 * instructionPercentCoverageChange : +0.31%
 * branchPercentCoverageChange : +0.57%
 * linePercentCoverageChange : +0.43%
 * complexityPercentCoverageChange : +0.62%
 * methodPercentCoverageChange : +0.31%
    
org.pentaho.di.job.entries.mailvalidator.JobEntryMailValidator:
 * instructionPercentCoverageChange : +0.10%
    
org.pentaho.di.repository.kdr.delegates.KettleDatabaseRepositoryPartitionSchemaDelegate:
 * instructionPercentCoverageChange :**-0.00%**
    
org.pentaho.di.job.entry.JobEntryCopy:
 * instructionPercentCoverageChange :**-0.03%**
    
org.pentaho.di.job.entries.fileexists.JobEntryFileExists:
 * instructionPercentCoverageChange : +0.26%
    
org.pentaho.di.trans.steps.textfileinput.TextFileInputMeta:
 * branchPercentCoverageChange : +0.53%
    
org.pentaho.di.job.entries.getpop.JobEntryGetPOP:
 * instructionPercentCoverageChange :**-0.00%**
    
org.pentaho.di.job.entries.pgpverify.JobEntryPGPVerify:
 * instructionPercentCoverageChange : +0.13%
    
org.pentaho.di.job.entries.filesexist.JobEntryFilesExist:
 * instructionPercentCoverageChange :**-0.07%**
    
org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * instructionPercentCoverageChange :**-0.28%**
 * branchPercentCoverageChange :**-3.33%**
 * complexityPercentCoverageChange :**-1.85%**
    
org.pentaho.di.trans.Trans:
 * instructionPercentCoverageChange :**-0.00%**
    
org.pentaho.di.job.entries.connectedtorepository.JobEntryConnectedToRepository:
 * instructionPercentCoverageChange : +0.13%
    
org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * instructionPercentCoverageChange :**-0.08%**
    
org.pentaho.di.trans.steps.randomvalue.RandomValueMeta:
 * instructionPercentCoverageChange :**-0.15%**
 * branchPercentCoverageChange :**-2.08%**
 * complexityPercentCoverageChange :**-2.13%**
    
org.pentaho.di.job.entries.webserviceavailable.JobEntryWebServiceAvailable:
 * instructionPercentCoverageChange : +0.17%
    
org.pentaho.di.trans.step.BaseStep:
 * instructionPercentCoverageChange : +0.01%
    
org.pentaho.di.job.entries.msgboxinfo.JobEntryMsgBoxInfo:
 * instructionPercentCoverageChange : +0.22%
    
org.pentaho.di.repository.kdr.KettleDatabaseRepository:
 * instructionPercentCoverageChange :**-0.00%**
    
org.pentaho.di.job.entries.talendjobexec.JobEntryTalendJobExec:
 * instructionPercentCoverageChange : +0.15%
    
org.pentaho.di.job.entries.checkdbconnection.JobEntryCheckDbConnections:
 * instructionPercentCoverageChange :**-0.02%**
    
org.pentaho.di.job.entries.sql.JobEntrySQL:
 * instructionPercentCoverageChange : +1.64%
 * linePercentCoverageChange : +1.97%
    
Integration test coverage change:
============
org.pentaho.di.trans.Trans:
 * instructionPercentCoverageChange :**-0.00%**
    
org.pentaho.di.trans.step.BaseStep:
 * instructionPercentCoverageChange : +0.04%
 * branchPercentCoverageChange : +0.30%
 * linePercentCoverageChange : +0.08%
 * complexityPercentCoverageChange : +0.18%

mbatchelor commented:
@deinspanjer - Could you review and merge please? Thanks.

deinspanjer commented:
I'm on it.

deinspanjer commented on core/src/org/pentaho/di/core/Const.java:
```diff
@@ -1357,6 +1357,30 @@ public static String rightPad( StringBuffer ret, int limit ) {
+   *          The desired length of the padded string.
+   * @return The padded String.
+   */
+  public static String rightPad( StringBuilder ret, int limit ) {
```
How about for all these copypasta functions, we just replace them with a single method that takes a CharSequence instead?  That way we can work with String, Buffer, and Builder inputs.

mbatchelor replied:
I checked into that. CharSequence doesn't have mutators. So, we'd not be able to mutate it unless we did "instanceof" checks and casts. I also checked their superclass (AbstractStringBuilder), but that's package-protected so I couldn't work with that either. I opted for copy/paste which I thought was better than "instanceof" stuff IMO.

deinspanjer replied:
I am so sorry if I'm being dense here, but I don't follow why you need to do any mutation or instanceof checks.  These rightPad methods and several similar sets such as isEmpty() and stripCR() all have a very similar pattern:

They take some instance of CharSequence (and frequently a limit parameter), and then they use a for loop to iterate the sequence and build the return which is either a new String or a boolean.

Instead of having three methods for each of these, could we not just have one method and use the charAt(), length(), and subSequence() methods of CharSequence to implement those for loops?

I agree that other areas where we are mutating a buffer, and for those, we don't want to try to unify the builder version with it.

mbatchelor commented:
isEmpty I agree with you 100% on. I was referring to the rPad method I added which (like the one receiving a StringBuffer) mutates the passed in StringBuffer/Builder. I acknowledge it returns a string, but you must also acknowledge that the passed in StringBuffer/Builder mutates as a result of the operation - so I needed to retain that behavior. That said, isEmpty could definitely be redone with CharSequence. 

So - I created a new JIRA case ( PDI-15007 ) to cover additional cleanup work (NPE/infinite loop fixes, and performance issues) in Const.java, and it is now in Commit 2 on this PR. In that, I changed isEmpty to receive a CharSequence, and I also cleaned up all the NPEs and infinite loops I mention on the new Jira case.
